dev/tests/integration/testsuite/Magento/Cms/Controller/Adminhtml/Wysiwyg/Images/DeleteFilesTest.php

```diff
@@ -75,22 +75,43 @@ protected function setUp()
      * Execute method with correct directory path and file name to check that files under WYSIWYG media directory
      * can be removed.
      *
+     * @param string $filename
      * @return void
+     * @dataProvider executeDataProvider
      */
-    public function testExecute()
+    public function testExecute(string $filename)
     {
+        $filePath =  $this->fullDirectoryPath . DIRECTORY_SEPARATOR . $filename;
+        $fixtureDir = realpath(__DIR__ . '/../../../../../Catalog/_files');
+        copy($fixtureDir . '/' . $this->fileName, $filePath);
+
         $this->model->getRequest()->setMethod('POST')
-            ->setPostValue('files', [$this->imagesHelper->idEncode($this->fileName)]);
+            ->setPostValue('files', [$this->imagesHelper->idEncode($filename)]);
         $this->model->getStorage()->getSession()->setCurrentPath($this->fullDirectoryPath);
         $this->model->execute();
 
         $this->assertFalse(
             $this->mediaDirectory->isExist(
-                $this->mediaDirectory->getRelativePath($this->fullDirectoryPath . '/' . $this->fileName)
+                $this->mediaDirectory->getRelativePath($this->fullDirectoryPath . '/' . $filename)
             )
         );
     }
 
+    /**
+     * DataProvider for testExecute
+     *
+     * @return array
+     */
+    public function executeDataProvider(): array
+    {
+        return [
+            ['magento_small_image.jpg'],
```

Comment:
I think the fix is also handling other special characters (not only the undersore), can you please add more variety to the data provider